### PR TITLE
Refactor bitesize1

### PR DESCRIPTION
**Pull request by t-young31**

### CODE REVIEW

File: tdms/include/field.h [hunk 1]
```diff
@@ -2,14 +2,44 @@
...
+ *    /                     /
+ *   /___________/_________/
+ *  (0, 0)     (1, 0)     (2, 0)
+ *
+ *  has I_tot = 2, J_tot = 1, K_tot = 0.
+ */
+class Grid{
+
+protected:
+    int I_tot = 0;
```
Comment (dstansby) [inline tdms/include/field.h:19]: Any reason not to go lower case on for ijk here? I'm worried that `I_tot` could also be interpreted as total electric current.

Comment (dstansby) [inline tdms/include/field.h:19]: Any reason not to go lower case on ijk here? I'm worried that `I_tot` could be interpreted as total electric current.

File: tdms/include/field.h [hunk 2]
```diff
@@ -2,14 +2,44 @@
 #include "simulation_parameters.h"
 
 
+/**
+ * A generic grid entity. For example:
```
Comment (dstansby) [inline tdms/include/field.h:6]: I'm quite confused by this. Is it a 3D grid? If so shouldn't the coordinates be three-tuples, not two-tuples?

File: tdms/include/field.h [hunk 3]
```diff
@@ -2,14 +2,44 @@
...
 struct xyz_arrays{
     double ***x;
     double ***y;
     double ***z;
 };
 
 
-class Field{
+/**
+ * A complex field defined over a grid. Has real and imaginary components
```
Comment (dstansby) [inline tdms/include/field.h]:
```suggestion
 * A complex vector field defined over a grid. Has three real and imaginary vector components
```
If I'm interpreting the code correctly?

File: tdms/src/fields/split.cpp [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        for (int i = 0; i < (I_tot + 1); i++) {
+          (*reinterpret_cast<double ****>(component))[k][j][i] = 0.;
+        }
+}
+
+SplitField::~SplitField() {
+
+  for (auto component : {xy, xz, yx, yz, zx, zy}){
+    if (component != nullptr && !has_no_elements()) {
+      destroy3DArray(reinterpret_cast<double ****>(component), J_tot+1, K_tot+1);
```
Comment (dstansby) [inline tdms/src/fields/split.cpp:35]: Is this missing an `I_tot+1`?

File: tdms/src/iterator.cpp [hunk 5]
```diff
@@ -6368,27 +6347,6 @@ if(runmode == rm_complete && (nvertices>0) )
...
-                           &J_s.zx, &J_s.zy,
-                           &J_nm1.xy, &J_nm1.xz,
-                           &J_nm1.yx, &J_nm1.yz,
-                           &J_nm1.zx, &J_nm1.zy);
-
-  }
-
-  //fprintf(stderr,"Pos 22\n");
-  if(is_cond){
-    destroy_auxilliary_mem_conductive(I_tot, J_tot, K_tot,
```
Comment (dstansby) [inline tdms/src/iterator.cpp:6387]: Are these destroys now taken care of elsewhere?

**Review by dstansby — APPROVED** [review]
This is a nice refactor! Some minor comments/questions, but looks 👍 overall

File: tdms/include/field.h (hunk 1 above)
Comment (t-young31) [reply tdms/include/field.h:19]: consistency I guess 🤷🏼 
definitely agree that it's a little confusing

File: tdms/include/field.h (hunk 2 above)
Comment (t-young31) [reply tdms/include/field.h:6]: they are 3 tuples(?!)

File: tdms/src/fields/split.cpp (hunk 4 above)
Comment (t-young31) [reply tdms/src/fields/split.cpp:35]: haha, no and I hate it

File: tdms/src/iterator.cpp (hunk 5 above)
Comment (t-young31) [reply tdms/src/iterator.cpp:6387]: yep – should be taken care of in the e.g. E_nm destructor

File: tdms/include/field.h (hunk 2 above)
Comment (dstansby) [reply tdms/include/field.h:6]: As in the coordinates should be e.g. (2, 1, 0), not (2, 1)

Comment (t-young31) [reply tdms/include/field.h:6]: ohh in the ascii – excellent point 🤦🏼

**Pull request merged**